src/AutoRest.CSharp/MgmtTest/Generation/Mock/MgmtMockTestBaseWriter.cs

```diff
@@ -211,27 +284,85 @@ protected void WriteTestOperation(CodeWriterDeclaration declaration, MockTestCas
             // we will always use the Async version of methods
             if (testCase.ClientOperation.IsPagingOperation)
             {
-                _writer.Append($"await foreach (var _ in ");
+                _writer.Append($"{(generatingSample ? $"" : $"await ")}foreach (var _ in ");
                 WriteTestMethodInvocation(declaration, testCase);
                 using (_writer.Scope($")"))
-                { }
+                {
+                    if (generatingSample && !testCase.ClientOperation.ReturnType.IsFrameworkType && testCase.ClientOperation.ReturnType.Implementation is Resource)
+                    {
+                        _writer.UseNamespace("System.Text.Json");
+                        _writer.Line($"Console.WriteLine($\"ResourceId: {{_.Data.Id}}\");");
+                        _writer.Line($"Console.WriteLine($\"Full Resource content: {{JsonSerializer.Serialize(_.Data)}}\");");
+                    }
+                }
             }
             else
             {
-                _writer.Append($"await ");
+                CodeWriterDeclaration operation = new CodeWriterDeclaration("operation");
+                if (generatingSample)
+                {
+                    _writer.Append($"var {operation:D} = ");
+                }
+                else
+                {
+                    _writer.Append($"await ");
+                }
                 WriteTestMethodInvocation(declaration, testCase);
                 _writer.LineRaw(";");
+                _writer.Line();
+                if (generatingSample)
+                {
+                    if (testCase.ClientOperation.ReturnType.Name == "ArmOperation")
+                    {
+                        _writer.UseNamespace("System.Text.Json");
+                        if (testCase.ClientOperation.ReturnType.Arguments.Length > 0 && testCase.ClientOperation.ReturnType.Arguments[0].Implementation is Resource)
+                        {
+                            _writer.Line($"Console.WriteLine($\"Succeed on ResourceId: {{{operation}.Value.Data.Id}}\");");
+                            _writer.Line($"Console.WriteLine($\"Full Resource content: {{JsonSerializer.Serialize({operation}.Value.Data)}}\");");
+                        }
+                        else
+                        {
+                            _writer.Line($"Console.WriteLine($\"Response: {{JsonSerializer.Serialize({operation}.Value)}}\");");
+                        }
+                    }
```

Comment:
We should have a flag on ClientOperation to tell you whether it is a LRO or not (including if this is a fake LRO). 
The idea is that if we have something high level, we will not depend on something low level (because low level things are complicated, processing them is usually a duplicated work and they are not as stable as the high level encapsulation)